macros/union_data.sql

```diff
@@ -92,16 +92,38 @@
     {%- endif -%}
 
 {%- else -%}
-    {%- set relation=adapter.get_relation(
-        database=var(database_variable, default_database),
-        schema=var(schema_variable, default_schema),
-        identifier=var(default_schema ~ '_' ~ table_identifier ~ '_' ~ 'identifier', table_identifier)) -%}
-
-{%- set table_exists=relation is not none -%}
+    {% set exception_schemas = {"linkedin_company_pages": "linkedin_pages", "instagram_business_pages": "instagram_business"} %}
+    {% set relation = namespace(value="") %}
+    {% if default_schema in exception_schemas.keys() %}
+        {% for corrected_schema_name in exception_schemas.items() %}   
+            {% if default_schema in corrected_schema_name %}
+                {# In order for this macro to effectively work within upstream integration tests (mainly used by the Fivetran dbt package maintainers), this identifier variable selection is required to use the macro with different identifier names. #}
+                {% set identifier_var = corrected_schema_name[1] + "_" + table_identifier + "_identifier"  %}
+                {%- set relation.value=adapter.get_relation(
+                    database=source(corrected_schema_name[1], table_identifier).database,
+                    schema=source(corrected_schema_name[1], table_identifier).schema,
+                    identifier=var(identifier_var, table_identifier)
+                ) -%}
+            {% endif %}
+        {% endfor %}
+    {% else %}
+        {# In order for this macro to effectively work within upstream integration tests (mainly used by the Fivetran dbt package maintainers), this identifier variable selection is required to use the macro with different identifier names. #}
+        {% set identifier_var = default_schema + "_" + table_identifier + "_identifier"  %}
+        {# Unfortunately the Twitter Organic identifiers were misspelled. As such, we will need to account for this in the model. This will be adjusted in the Twitter Organic package, but to ensure backwards compatibility, this needs to be included. #}
+        {% if var(identifier_var, none) is none %} 
+            {% set identifier_var = default_schema + "_" + table_identifier + "_identifer"  %}
+        {% endif %}
+        {%- set relation.value=adapter.get_relation(
```

Comment:
also why relation.value instead of relation?

Reply:
Great question, I had to use namespaces in order to ensure the relation value would be accessible outside of a for loop. Since we set the namespace variable [here](https://github.com/fivetran/dbt_fivetran_utils/blob/6fd189a87eeb6370b1697e562fdac6b15e288270/macros/union_data.sql#L96), we then need to access the value by specifying it in the variable call.

Reply:
ahh right the namespace! good to know